### PR TITLE
Replace LoadingCache usage with a simple ConcurrentHashMap

### PR DESCRIPTION
**Pull request by s1monw**
This commit replaces the usage of LoadedCache with a simple CHM and calls
to computeIfAbsent and adds LoadingCache and CacheLoader to forbidden APIs

Relates to #13224

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/common/LazyLoadedKeyedSupplier.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+     * This method will not load a vlaue for the given key
+     */
+    public V tryGet(K key) {
+        return delegate.get(key);
+    }
+
+    /**
+     * Looksup or loads a value for the given key.
+     */
+    public V get(K key) {
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/common/LazyLoadedKeyedSupplier.java]: I think [`ConcurrentHashMap#computeIfAbsent`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#computeIfAbsent-K-java.util.function.Function-) will help here. :)

**Comment by s1monw** [comment]
@jasontedor I simplified it :)

**Comment by jasontedor** [comment]
LGTM.

File: core/src/main/java/org/elasticsearch/indices/analysis/HunspellService.java [hunk 2]
```diff
@@ -71,24 +70,30 @@
...
-        this.knownDictionaries = knownDictionaries;
+        this.knownDictionaries = Collections.unmodifiableMap(knownDictionaries);
         this.hunspellDir = resolveHunspellDirectory(settings, env);
         this.defaultIgnoreCase = settings.getAsBoolean(HUNSPELL_IGNORE_CASE, false);
-        this.env = env;
+        this.loadingFunction = (locale) -> {
+            try {
+                return loadDictionary(locale, settings, env);
+            } catch (Throwable e) {
+                throw new IllegalStateException("failed to load hunspell dictionary for local: " + locale, e);
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/indices/analysis/HunspellService.java]: Minor typo of `local` instead of `locale` in the exception message.

**Comment by s1monw** [comment]
alright I wil push this now

**Comment by nik9000** [comment]
I haven't reviewed the code but I imagine this is probably worth
benchmarking some.

**Comment by jasontedor** [comment]
@nik9000 This isn't the big caching replacement.

**Comment by s1monw** [comment]
@nik9000 I don't think we need to benchmark the hunspell dictionary lazyloading cache

**Comment by nik9000** [comment]
> @nik9000 This isn't the big caching replacement.

Thanks! I really was just scanning through my email this morning I hadn't got to check it yet.

**Comment by nik9000** [comment]
And I just reviewed and agree with pushing it. No benchmarking required. I withdraw that.

**Pull request merged**